pkl-core/src/test/files/LanguageSnippetTests/input/types/typeAlias2.pkl

```diff
@@ -35,3 +35,10 @@ typealias Bar = Foo
 class Baz extends Bar {}
 
 res7 = (new Baz {}).value
+
+typealias MappingOrListing = Mapping|Listing
+
+res8 = new {
+  foo = 5
+  bar = if (foo is MappingOrListing) "bar" else foo
```

Comment:
This is failing in the main branch because the type check on `MappingOrListing` (`aliasedTypeNode.execute(...)` throws a (caught) type check exception, thus the owner/receiver don't get reset. Pkl then executes the `else` branch, and the owner/receiver are incorrectly set to the module.

Reply:
Good catch